pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/elastic/ElasticSearchService.java

```diff
@@ -0,0 +1,103 @@
+package org.broadinstitute.dsm.service.elastic;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.dto.ddp.participant.ParticipantData;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.model.elastic.ObjectTransformer;
+import org.broadinstitute.dsm.model.elastic.Profile;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearch;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearchParticipantDto;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+/**
+ * Service for interacting with ElasticSearch
+ * </p>
+ * Note: This is based on org.broadinstitute.dsm.model.elastic.search.ElasticSearch, and we move methods from there
+ * as they get untangled from ElasticSearchUtil and come under automated testing
+ */
+@Slf4j
+public class ElasticSearchService {
+    private final ElasticSearch elasticSearch;
+
+    public ElasticSearchService() {
+        this.elasticSearch = new ElasticSearch();
+    }
+
+    /**
+     * Return a list of all participant guids in a participant ES index
+     */
+    public List<String> getAllParticipantGuids(String esParticipantsIndex) {
+        ElasticSearch es = elasticSearch.getAllParticipantsDataByInstanceIndex(esParticipantsIndex);
+        List<String> participantIds = new ArrayList<>();
+        for (ElasticSearchParticipantDto participantDto : es.getEsParticipants()) {
+            if (participantDto.getProfile().isPresent()) {
+                Profile profile = participantDto.getProfile().get();
+                if (StringUtils.isNotBlank(profile.getGuid())) {
```

Comment:
Should we be checking for guid uniqueness and removing/logging duplicate participant guids?

Reply:
One thing that I think would help reviewers is a PR comment about what methods are being moved, on both sides of the refactor (the original class with the method, and the new method's home).  Without this information, the reviewer must do their own analysis to determine whether the "new" method is actually new, or just different, and that's often time-consuming.  When it looks like it _might_ be different, I end up making suggestions like this, which are probably a distraction at best.  I haven't seen cases where we have duplicate guids, so I'm okay just leaving this as-is until we have a signal that shows otherwise.

Reply:
Totally agree. My bad for not doing that here.